runtime/vm/resolvesupport.cpp

```diff
@@ -2105,56 +2260,80 @@ resolveInvokeDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA callSite
 {
 	Assert_VM_true(J9_RESOLVE_FLAG_RUNTIME_RESOLVE == resolveFlags);
 	j9object_t *callSite = ramCP->ramClass->callSites + callSiteIndex;
-	j9object_t methodHandle = *callSite;
+	j9object_t result = *callSite;
 
 	J9ROMClass *romClass = ramCP->ramClass->romClass;
 	J9SRP *callSiteData = (J9SRP *) J9ROMCLASS_CALLSITEDATA(romClass);
 	U_16 *bsmIndices = (U_16 *) (callSiteData + romClass->callSiteCount);
 	U_16 *bsmData = bsmIndices + romClass->callSiteCount;
 	J9ROMNameAndSignature* nameAndSig = SRP_PTR_GET(callSiteData + callSiteIndex, J9ROMNameAndSignature*);
 	U_16 bsmIndex = bsmIndices[callSiteIndex];
-	U_16 i;
+	U_16 i = 0;
 
 	/* Check if already resolved */
-	if (methodHandle != NULL) {
-		return methodHandle;
+	if (result != NULL) {
+		return result;
```

Comment:
If possible, avoid multiple return statements. This will make it easier to position entry and exit tracepoints.